### ISSUE TITLE
TextInput with suggestions breaks keyboard accessibility when used in a modal Layer

### ISSUE THREAD

**Issue by eatyourgreens** [grommet/grommet#5260]
We are using `TextInput` with suggestions inside a modal popup, which uses `Layer`. After selecting a suggestion, then closing the popup, we find that you can no longer Tab through the parent page. All focusable inputs, buttons, links etc. have `tabindex=-1`.

### Expected Behavior

Keyboard Tab navigation should be cancelled while the `Layer` is in place, but should be enabled again once it's removed.

### Actual Behavior

The `Layer` sets `data-g-tabindex='none' tabindex=-1` on each focusable element when it opens. When the text suggestions open, the `Drop` component seems to update those to `data-g-tabindex=-1 tabindex=-1`, which prevents the tabindex from ever being removed from the element again.

### URL, screen shot, or Codepen exhibiting the issue

Here's the link to the related issue on our project, including a link with instructions to reproduce the bug.
https://github.com/zooniverse/front-end-monorepo/issues/2183

I think the bug might be in the DOM library here, where the `TABINDEX_STATE` attribute is written without checking that it already exists (ie. that we're already in an open modal.)
https://github.com/grommet/grommet/blob/8a8ef34f119bf6dd3b5768436257d203e7c7f5fa/src/js/utils/DOM.js#L134-L138

- Grommet version:  2.17.1.
- Browser Name and version: Firefox and Chrome latest.
- Operating System and version (desktop or mobile): MacOS Mojave.

**Comment by ShimiSun** [comment]
This issue might be related to grommet-icons more than grommet.

**Comment by jcfilben** [comment]
Resolved by https://github.com/grommet/grommet/pull/7324
